app/assets/javascripts/application.js

```diff
@@ -1 +1,21 @@
-//= require header-footer-only
+/* global $ */
+
+//= require govuk_publishing_components/dependencies
+
+//= require govuk_publishing_components/lib
+//= require govuk_publishing_components/lib/track-click
+
+//= require analytics
+
+//= require govuk_publishing_components/components/cookie-banner
+//= require govuk_publishing_components/components/layout-header
+//= require govuk_publishing_components/components/feedback
+//= require govuk_publishing_components/modules
+
+//= require modules/global-bar
+//= require modules/cross-domain-tracking
+
+//= require global-bar-init
```

Comment:
Interestingly `global-bar-init` includes the cookie functions code from the gem, which is already pulled in by `analytics.js` in the gem, which is included here on line 8. Again, possibly out scope for this PR, but I wonder if there's an optimisation to made here, or whether rails is smart enough to not duplicate this code?

Reply:
I don't know if there's any de-duping or tree-shaking going on in Rails's asset pipeline. This is the kind of thing that Webpack would help with - we could include dependencies as many times as we want and Webpack would ensure that they're only in the compiled code once.